constraints.md

```diff
@@ -1,192 +1,198 @@
----
-title: Constraints
-summary: Learn how SQL Constraints apply to TiDB.
-category: reference
-aliases: ['/docs/dev/reference/sql/constraints/']
----
```

Comment:
Please don't remove the header! It sets the title and adds redirects from previous file locations.